ledger-service/http-json/src/test/scala/com/digitalasset/http/endpoints/MeteringReportEndpointTest.scala

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+package endpoints
+
+import com.daml.ledger.api.v1.admin.metering_report_service
+import com.daml.lf.data.Ref.{ApplicationId, ParticipantId}
+import com.daml.lf.data.Time.Timestamp
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import scalaz.\/-
+import spray.json.enrichAny
+
+class MeteringReportEndpointTest extends AnyFreeSpec with Matchers {
+
+  import MeteringReportEndpoint._
+
+  "MeteringReportEndpoint" - {
+
+    val appX = ApplicationId.assertFromString("appX")
+
+    val request = MeteringReportRequest(
+      Timestamp.assertFromString("2022-02-03T14:00:00Z"),
+      Some(Timestamp.assertFromString("2022-02-03T15:00:00Z")),
+      Some(appX),
+    )
+
+    val report = MeteringReport(
+      ParticipantId.assertFromString("Part"),
+      request.copy(to = None),
+      `final` = true,
+      Seq(ApplicationMeteringReport(appX, 63)),
+    )
+
+    "should read/write request" in {
+      val actual = request.toJson.convertTo[MeteringReportRequest]
+      actual shouldBe request
+    }
+
+    "should read/write report" in {
+      val json = report.toJson
+      println(json)
+      val actual = json.convertTo[MeteringReport]
+      actual shouldBe report
+    }
+
+    "should convert to timestamp to protobuf timestamp" in {
+      val expected = Timestamp.assertFromString("2022-02-03T14:00:00Z")
+      val actual = toTimestamp(toPbTimestamp(expected))
+      actual shouldBe expected
+    }
+
+    "should convert to protobuf request" in {
+      import request._
+      val expected = metering_report_service.GetMeteringReportRequest(
+        Some(toPbTimestamp(from)),
+        to.map(toPbTimestamp),
+        application.get,
+      )
+      val actual = MeteringReportEndpoint.toPbRequest(request)
+      actual shouldBe expected
+    }
+
+    "default protobuf strings are empty string" in {
+      metering_report_service.GetMeteringReportRequest.defaultInstance.applicationId shouldBe ""
+    }
```

Comment:
I don't think we should test the libraries and/or generated code we use, but I understand where you're coming from. Can you perhaps to change this test to make sure that we deal with empty strings correctly (e.g. using your `pbOption` method), perhaps (but not necessarily) in conjunction with why this is important in the broader scope of how the string is used.

Perhaps a better test could be to convert a Protobuf object with an empty string in it and see that it's correctly converted into a `None`. If this test is already there and I missed it, then probably this test case is not particularly helpful.

Reply:
@stefanobaghino-da - great comments, fill fix up tomorrow.

Reply:
That was added as I can never remember if proto3 uses and empty string or null by default.  Removed.